jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/ContextHandlerLifeCycle.java

```diff
@@ -22,24 +22,25 @@
 
 import org.eclipse.jetty.deploy.graph.Graph;
 import org.eclipse.jetty.deploy.graph.Node;
+import org.eclipse.jetty.server.handler.ContextHandler;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * The lifecycle of an App in the {@link DeploymentManager}.
+ * The lifecycle of a {@link ContextHandler} in the {@link DeploymentManager}.
  * <p>
- * Setups a the default {@link Graph}, and manages the bindings to the life cycle via the {@link AppLifeCycle.Binding}
+ * Sets up the default {@link Graph}, and manages the bindings to the life cycle via the {@link ContextHandlerLifeCycle.Binding}
  * annotation.
  * <p>
- * <img alt="app lifecycle graph" src="doc-files/AppLifeCycle.png">
+ * <img alt="context-handler lifecycle graph" src="doc-files/ContextHandlerLifeCycle.png">
  */
-public class AppLifeCycle extends Graph
+public class ContextHandlerLifeCycle extends Graph
```

Comment:
To play devils advocate with the design.  Do we really need a fully extensible state machine for deployment?       We already have events from our standard o.e.j.util.component of:
 + org.eclipse.jetty.util.component.Container.Listener.beanAdded
 + org.eclipse.jetty.util.component.Container.Listener.beanRemoved
 + org.eclipse.jetty.util.component.LifeCycle.Listener.lifeCycleStarting
 + org.eclipse.jetty.util.component.LifeCycle.Listener.lifeCycleStarted
 + org.eclipse.jetty.util.component.LifeCycle.Listener.lifeCycleFailure
 + org.eclipse.jetty.util.component.LifeCycle.Listener.lifeCycleStopping
 + org.eclipse.jetty.util.component.LifeCycle.Listener.lifeCycleStopped

So perhaps we can just have a fixed state machine based on those events and the ability to bind to those instead.